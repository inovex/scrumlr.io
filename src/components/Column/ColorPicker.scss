@import "src/constants/style";

.column__header-color-picker-dropdown {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: $spacing--xxs;
  padding-top: $spacing--xxs;
  padding-bottom: $spacing--xxs;

  width: 40px;

  background-color: $gray--300;
  //box-shadow:
  //  0 4px 6px -1px rgb(0 0 0 / 0.1),
  //  0 2px 4px -2px rgb(0 0 0 / 0.1);
  list-style: none;
  //padding: 10px;
  //padding-top: 30px;
  padding-left: 0;
  margin-top: auto;
  border-radius: 24px;
  //position: inherit;
  //left: 30px;

  &:hover {
    background-color: $brighten--slightly;
  }
}

.column__header-color-picker-dropdown > li > button {
  cursor: pointer;
  border: none;
  background: none;
  padding: 0;
  width: 32px;
}

.column__header-color-picker-dropdown > li {
  all: unset;
  display: flex;
  justify-content: center;
  align-items: center;
  //width: 40px;
  //height: 40px;
  cursor: pointer;
}

.column__header-color-picker-dropdown > ul > li {
  padding: 0;
}

[theme="dark"] {
  .column__header-color-picker-dropdown {
    background-color: $navy--500;
  }

  .planning-pink {
    border: $navy--500;
  }
}

.column__header-color-option {
  width: 23px;
  height: 23px;
  border-radius: 100px;
}

.planning-pink {
  background: #ff3387;
  border: 3px solid $gray--300;
}

.planning-pink_selected {
  background: #ff3387;
  border: 3px solid #ff99c3;
}

.planning-pink:hover {
  border: 3px solid #ffcce1;
}

.backlog-blue {
  background: #3379ff;
  border: 3px solid $gray--300;
}

.backlog-blue_selected {
  background: #3379ff;
  border: 3px solid #99bcff;
}

.backlog-blue:hover {
  border: #ccddff;
}

.backlog-blue:hover {
  border: 3px solid #ccddff;
}

.poker-purple {
  background: #d44fff;
  border: 3px solid $gray--300;
}

.poker-purple_selected {
  background: #d44fff;
  border: 3px solid #e9a7ff;
}

.poker-purple:hover {
  border: 3px solid #f4d3ff;
}

.value-violet {
  background: #7d33f8;
  border: 3px solid $gray--300;
}

.value-violet_selected {
  background: #7d33f8;
  border: 3px solid #be99fb;
}

.value-violet:hover {
  border: 3px solid #dfccfd;
}

.goal-green {
  background: #49dd98;
  border: 3px solid $gray--300;
}

.goal-green_selected {
  background: #49dd98;
  border: 3px solid #a4eecb;
}

.goal-green:hover {
  border: 3px solid #d1f6e5;
}

.yielding-yellow {
  background: #fff733;
  border: 3px solid $gray--300;
}

.yielding-yellow_selected {
  background: #fff733;
  border: 3px solid #fffb99;
}

.yielding-yellow:hover {
  border: 3px solid #fffdcc;
}

.online-orange {
  background: #ffb933;
  border: 3px solid $gray--300;
}

.online-orange_selected {
  background: #ffb933;
  border: 3px solid #ffdc99;
}

.online-orange:hover {
  border: 3px solid #ffeecc;
}
