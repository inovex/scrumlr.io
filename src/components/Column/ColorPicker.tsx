import React from "react";
// import {types} from "sass";
// import Color = types.Color;
import classNames from "classnames";
import {useDispatch} from "react-redux";
import {getColorClassName} from "../../constants/colors";
import {Actions} from "../../store/action";

// Definieren Sie die Schnittstelle f√ºr die Props der Komponente
export interface ColorPickerProps {
  id: string;
  name: string;
  visible: boolean;
  index: number;
}

// Definieren Sie die Komponente
export const ColorPicker: React.FC<ColorPickerProps> = ({id, name, visible, index}) => {
  const dispatch = useDispatch();

  return (
    <ul className="column__header-color-picker-dropdown">
      <li>
        <button
          aria-label="Planning Pink"
          title="Planning Pink"
          // className={classNames(getColorClassName("planning-pink"), "column__color-button")}
          onClick={() => dispatch(Actions.editColumn(id, {name, color: "planning-pink", index, visible}))}
          // title={t("Column.settings")} className="column__header-edit-button"
        >
          <div className="column__header-color-option planning-pink" />
        </button>
      </li>
      <li>
        {/* cool */}
        <button
          aria-label="Backlog Blue"
          title="Backlog Blue"
          className={classNames(getColorClassName("backlog-blue"), "column__color-button")}
          onClick={() => dispatch(Actions.editColumn(id, {name, color: "backlog-blue", index, visible}))}
        >
          <div className="column__header-color-option backlog-blue" />
        </button>
      </li>
      <li>
        {/* cool */}
        <button
          aria-label="Grooming Green"
          title="Grooming Green"
          className={classNames(getColorClassName("grooming-green"), "column__color-button")}
          onClick={() => dispatch(Actions.editColumn(id, {name, color: "grooming-green", index, visible}))}
        >
          <div className="column__header-color-option grooming-green" />
        </button>
      </li>
      <li>
        {/* cool */}
        <button
          aria-label="Poker Purple"
          title="Poker Purple"
          className={classNames(getColorClassName("poker-purple"), "column__color-button")}
          onClick={() => dispatch(Actions.editColumn(id, {name, color: "poker-purple", index, visible}))}
        >
          <div className="column__header-color-option poker-purple" />
        </button>
      </li>
      <li>
        {/* cool */}
        <button
          aria-label="Online Orange"
          title="Online Orange"
          className={classNames(getColorClassName("online-orange"), "column__color-button")}
          onClick={() => dispatch(Actions.editColumn(id, {name, color: "online-orange", index, visible}))}
        >
          <div className="column__header-color-option online-orange" />
        </button>
      </li>
      <li>
        {/* cool */}
        <button
          aria-label="Lean Lilac"
          title="Lean Lilac"
          className={classNames(getColorClassName("lean-lilac"), "column__color-button")}
          onClick={() => dispatch(Actions.editColumn(id, {name, color: "lean-lilac", index, visible}))}
        >
          <div className="column__header-color-option lean-lilac" />
        </button>
      </li>
      <li>
        {/* cool */}
        <button
          aria-label="Retro Red"
          title="Retro Red"
          className={classNames(getColorClassName("retro-red"), "column__color-button")}
          onClick={() => dispatch(Actions.editColumn(id, {name, color: "retro-red", index, visible}))}
        >
          <div className="column__header-color-option retro-red" />
        </button>
      </li>
    </ul>
  );
};
