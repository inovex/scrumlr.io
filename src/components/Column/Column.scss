@import "constants/style";

.column {
  min-width: $column__min-width;
  width: 100%;
  max-width: $column__max-width;
  @include inset-border($top: true, $bottom: true);
  padding-top: $header__height;
  height: 100vh;
}
.column:nth-child(even) {
  background-color: #f9fafb;
}
.column:nth-child(odd) {
  background-color: #f6f5fb;
}
@media (prefers-color-scheme: dark) {
  .column:nth-child(even) {
    background-color: #242c3d;
  }
  .column:nth-child(odd) {
    background-color: #272f41;
  }
}

.column__content {
  max-height: 100%;
  padding: $padding--large 0 $column__border-width 0;
  display: flex;
  flex-direction: column;
}

.column__header {
  padding: 0 $padding--large;
}

.column__header-title {
  display: flex;
}

.column__header-text {
  font-family: Raleway;
  font-size: 24px;
  color: $color-black;
  border-bottom: solid 3px var(--accent-color);
  margin-right: 10px;
  font-weight: bold;
  letter-spacing: $letter-spacing--large;
  line-height: $line-height--large;
}
@media (prefers-color-scheme: dark) {
  .column__header-text {
    color: $color-white;
  }
}

.column__header-card-number {
  font-family: Raleway;
  font-size: 20px;
  color: $color-middle-gray;
  font-weight: 600;
  letter-spacing: $letter-spacing--large;
  line-height: $line-height--large;
}

.column__notes-wrapper {
  overflow: auto;
}

.column__notes-wrapper::-webkit-scrollbar {
  width: 12px;
}

.column__notes-wrapper::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px $color-light-gray;
  border-radius: 10px;
}

.column__notes-wrapper::-webkit-scrollbar-thumb {
  border-radius: 10px;
  background-color: $color-middle-gray;
}

@media (prefers-color-scheme: dark) {
  .column__notes-wrapper::-webkit-scrollbar-track {
    -webkit-box-shadow: inset 0 0 6px $color-dark-mode;
    background-color: $color-dark-mode-note;
  }
  .column__notes-wrapper::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: $color-darkest-gray;
  }
}

.column__note-list {
  margin: 0;
  padding: 0 $padding--large $padding--default $padding--large;
  display: grid;
  grid-auto-flow: row;
  grid-row-gap: 20px;
  grid-column-gap: 24px;
}
