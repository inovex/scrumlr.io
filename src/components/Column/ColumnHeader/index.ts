import ColumnHeader from './ColumnHeader';
export default ColumnHeader;
