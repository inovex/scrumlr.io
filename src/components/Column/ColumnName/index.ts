import ColumnName from './ColumnName';
export default ColumnName;
