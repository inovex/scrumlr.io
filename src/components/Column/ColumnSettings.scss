@import "src/constants/style";

.column__header-menu-dropdown {
  position: absolute;
  right: 0;
  top: 75%;
  width: 180px;
  background-color: $color-white;
  border-radius: 8px;
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1);
  z-index: 150;
}

.column__header-menu-dropdown > ul {
  list-style: none;
  padding: $padding--small 0;
  margin: 0;
}

.column__header-menu-dropdown > ul > li {
  height: 32px;
  padding: 0 16px;
  background-color: $color-white;

  &:hover {
    filter: $darken--slightly;
  }

  &:focus-within {
    background-color: $color-white-two;
  }
}

.column__header-menu-dropdown > ul > li > button {
  cursor: pointer;
  border: 0;
  outline: none;
  background-color: transparent;
  padding: 0;
  margin: 0;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  gap: 10px;
  font-size: $text-size--small;
}

.column__header-menu-dropdown > ul > li > button > svg {
  height: $icon--medium;
  width: $icon--medium;
}

.column__header-menu-dropdown > ul > li:last-of-type {
  height: auto;
  padding: $padding--small $padding--default;
  display: flex;
  justify-content: space-between;
}

.column__header-menu-dropdown > ul > li:last-child > button {
  height: 18px;
  width: 18px;
  border-radius: 4px;
  background-color: var(--accent-color);
  transition: all 0.08s ease-out;

  &:hover {
    transform: scale(1.1);
  }

  &:focus-visible {
    box-shadow: 0 0 0 2px $color-darker-gray;
  }
}

// Hacky way to not show delete column button if it's the last column
.column:only-of-type .column__header-menu-dropdown > ul > li:nth-child(5) {
  display: none;
}

[theme="dark"] {
  .column__header-menu-dropdown {
    background-color: $color-dark-mode-note;
  }

  .column__header-menu-dropdown > ul > li {
    background-color: $color-dark-mode-note;

    &:hover {
      filter: $brighten--slightly;
    }

    &:focus-within {
      background-color: $color-dark-mode-note--hover;
    }
  }

  .column__header-menu-dropdown > ul > li > button {
    color: $color-white;
  }

  .column__header-menu-dropdown > ul > li:last-child > button:focus-visible {
    box-shadow: 0 0 0 2px $color-white;
  }
}
