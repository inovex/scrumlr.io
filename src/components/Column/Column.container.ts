import { BoardCards, Card, StoreState } from '../../types';
import { getFirebase, getVal } from 'react-redux-firebase';
import { OwnColumnProps, StateColumnProps } from './Column';
import {
  ColumnType,
  getTheme,
  PhaseConfiguration
} from '../../constants/Retrospective';
import { Key } from 'ts-keycode-enum';
import { get } from 'lodash';
import Raven = require('raven-js');

function sortCards(cards: Card[], sortByVotes: boolean): Card[] {
  const compareTimestamps = (a: Card, b: Card) => {
    if (a.timestamp > b.timestamp) {
      return 1;
    } else if (a.timestamp < b.timestamp) {
      return -1;
    } else {
      return 0;
    }
  };

  if (sortByVotes) {
    return cards
      .sort((a, b) => {
        if (a.votes > b.votes) {
          return 1;
        } else if (a.votes < b.votes) {
          return -1;
        } else {
          return compareTimestamps(a, b);
        }
      })
      .reverse();
  } else {
    return cards.sort(compareTimestamps).reverse();
  }
}

const getColumnType = (a: string, config: PhaseConfiguration): ColumnType => {
  if (typeof a === 'undefined') {
    return 'positive';
  }

  if (isNaN(a as any)) {
    return a as ColumnType;
  }
  return config.columns[parseInt(a)].type;
};

const isEqualColumnType = (
  a: string,
  b: string,
  config: PhaseConfiguration
) => {
  if (typeof a === 'undefined' || typeof b === 'undefined') {
    return false;
  }

  let aMapped = a;
  if (isNaN(a as any)) {
    const columns = Object.keys(config.columns)
      .filter(key => config.columns[key].id === a)
      .map(key => key);
    if (columns.length === 1) {
      aMapped = columns[0];
    }
  }

  let bMapped = b;
  if (isNaN(b as any)) {
    const columns = Object.keys(config.columns)
      .filter(key => config.columns[key].id === b)
      .map(key => key);
    if (columns.length === 1) {
      bMapped = columns[0];
    }
  }

  return aMapped === bMapped;
};

export const mapStateToProps = (
  state: StoreState,
  ownProps: OwnColumnProps
): StateColumnProps => {
  const user = getVal(state.fbState, 'auth', undefined);
  const isAdmin =
    getVal(
      state.fbState,
      `data/${ownProps.boardUrl}/config/creatorUid`,
      undefined
    ) === user.uid;

  const boardCards: BoardCards = getVal(
    state.fbState,
    `data/${ownProps.boardUrl}/cards`,
    {}
  );

  const focusedCardId: string = getVal(
    state.fbState,
    `data/${ownProps.boardUrl}/config/focusedCardId`,
    undefined
  );

  if (!focusedCardId) {
    window.onkeydown = () => {};
  }

  let focusTarget = undefined;

  const focusedCard = boardCards[focusedCardId];
  let focused: Card | undefined = undefined;

  if (focusedCard) {
    if (
      isEqualColumnType(focusedCard.type, ownProps.id, ownProps.phase.config)
    ) {
      focused = focusedCard;
      focused.id = focusedCardId;
    }
    focusTarget = get(
      ownProps.phase.config.columns[focusedCard.type],
      'focus.column'
    );
  }

  const showAsFocusTarget = focusedCard ? ownProps.id === focusTarget : false;
  const isFocusOrigin = focusedCard
    ? isEqualColumnType(focusedCard.type, ownProps.id, ownProps.phase.config)
    : false;

  const isHidden = focusedCard ? !isFocusOrigin && !showAsFocusTarget : false;

  const isExtended = focusedCard
    ? isFocusOrigin && ownProps.id !== focusTarget
    : false;

  let cards: Card[] = Object.keys(boardCards)
    .map(key => {
      return { id: key, ...boardCards[key] };
    })
    .filter(
      card =>
        ownProps.phase.guidedPhase !== 0 ||
        ownProps.isShowCards ||
        card.authorUid === user.uid
    )
    .filter(card =>
      isEqualColumnType(card.type, ownProps.id, ownProps.phase.config)
    )
    .filter(card => !Boolean(card.parent));

  cards = sortCards(cards, ownProps.phase.config.columns[ownProps.id].sorted);

  let cardsWithFocused: Card[] = Object.keys(boardCards)
    .map(key => {
      return { id: key, ...boardCards[key] };
    })
    .filter(card =>
      isEqualColumnType(card.type, ownProps.id, ownProps.phase.config)
    )
    .filter(
      card =>
        !Boolean(card.parent) || (focused && card.id === (focused as Card).id)
    );
  cardsWithFocused = sortCards(
    cardsWithFocused,
    ownProps.phase.config.columns[ownProps.id].sorted
  );

  function onFocusCard(cardId: string | null) {
    getFirebase()
      .ref(`${ownProps.boardUrl}/config/focusedCardId`)
      .set(cardId)
      .catch((err: Error) => {
        Raven.captureMessage('Could not set focus on card', {
          extra: {
            reason: err.message,
            uid: user.uid,
            boardId: ownProps.boardUrl,
            cardId
          }
        });
      });
  }

  if (
    focused &&
    isEqualColumnType(focused.type, ownProps.id, ownProps.phase.config)
  ) {
    const isFocusedStacked = cards.length !== cardsWithFocused.length;
    const stackToSearch = isFocusedStacked ? cardsWithFocused : cards;

    let index = stackToSearch.findIndex((card: Card) => {
      return card.id === (focused as Card).id;
    });

    if (state.app.keyboardNavigationEnabled) {
      window.onkeydown = (ev: KeyboardEvent) => {
        if (isAdmin) {
          // on escape
          if (ev.keyCode === Key.Escape) {
            if (document.getElementById('portal')!.children.length === 0) {
              onFocusCard(null);
              ev.preventDefault();
            }
          }

          // on arrow up
          if (ev.keyCode === Key.UpArrow) {
            if (index > 0) {
              onFocusCard(cards[index - 1].id as string);
              ev.preventDefault();
            }
          }

          // on arrow down
          if (ev.keyCode === Key.DownArrow) {
            if (index < cards.length - 1) {
              onFocusCard(cards[index + (isFocusedStacked ? 0 : 1)]
                .id as string);
              ev.preventDefault();
            }
          }
        }
      };
    } else {
      window.onkeydown = () => {};
    }
  }

  const theme = getTheme(
    getColumnType(
      ownProps.phase.config.columns[ownProps.id].type,
      ownProps.phase.config
    )
  );

  return {
    theme,
    cards,
    focused,
    isHidden,
    isExtended
  };
};
