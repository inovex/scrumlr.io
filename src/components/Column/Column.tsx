import * as classNames from 'classnames';
import * as React from 'react';

import './Column.scss';
import { Card as CardModel } from '../../types';
import {
  ColumnConfiguration,
  IndexedPhaseConfiguration
} from '../../constants/Retrospective';
import { mapStateToProps } from './Column.container';
import FocusedCardComponent from './FocusedCardComponent';
import StackComponent from './StackComponent';
import ColumnHeader from './ColumnHeader';
import * as ReactCSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
import { Icon } from '../Icon';
import { connect } from 'react-redux';
import { ColumnOverview } from '../ColumnOverview';

export interface OwnColumnProps {
  boardUrl: string;
  isAdmin: boolean;

  /** The unique column type. */
  column: ColumnConfiguration;
  phase: IndexedPhaseConfiguration;

  isActive: boolean;

  hasNextColumn?: boolean;
  hasPreviousColumn?: boolean;
  onGoToNextColumn?: () => void;
  onGoToPrevColumn?: () => void;
  isCompactView: boolean;

  className?: string;
}

export interface StateColumnProps {
  theme: string;
  cards: CardModel[];
  focused?: CardModel;
  isHidden: boolean;
  isExtended: boolean;
}

export type ColumnProps = OwnColumnProps & StateColumnProps;

export interface ColumnState {
  showOverview: boolean;
}

export class Column extends React.Component<ColumnProps, ColumnState> {
  state: ColumnState = {
    showOverview: false
  };

  toggleOverview = () => {
    this.setState({ showOverview: !this.state.showOverview });
  };

  render() {
    const {
      isAdmin,
      isActive,
      isHidden,
      isExtended,
      className,
      focused,
      column,
      phase,
      theme,
      boardUrl,
      cards,
      isCompactView
    } = this.props;

    const cardsCount = cards.filter(c => !c.parent).length;

    return (
      <>
        {this.state.showOverview && (
          <ColumnOverview
            boardUrl={boardUrl}
            column={column.name}
            cardsCount={cardsCount}
            cards={cards}
            isVotingEnabled={column.voting.enabled}
            isVoteSummaryShown={column.voting.displayed}
            toggleOverview={this.toggleOverview}
          />
        )}
        <div
          className={classNames('column', `column--theme-${theme}`, className, {
            ['column--inactive']:
              !isActive && (focused ? focused.type !== column.id : true),
            ['column--hidden']: isHidden
          })}
          style={{ flex: isExtended ? phase.columns.length - 1 : 1 }}
        >
          {!focused && this.props.hasPreviousColumn && (
            <button
              className={classNames(
                'column__navigation',
                'column__navigation-left'
              )}
              type="button"
              onClick={this.props.onGoToPrevColumn}
              disabled={!isActive}
            >
              <Icon name="chevron-left" width={32} height={32} />
            </button>
          )}

          {!focused && this.props.hasNextColumn && (
            <button
              className={classNames(
                'column__navigation',
                'column__navigation-right'
              )}
              type="button"
              onClick={this.props.onGoToNextColumn}
              disabled={!isActive}
            >
              <Icon name="chevron-right" width={32} height={32} />
            </button>
          )}

          <ColumnHeader
            title={column.name}
            count={cardsCount}
            onToggleOverview={this.toggleOverview}
          />

          <ReactCSSTransitionGroup
            transitionName="column__content-animation"
            transitionEnterTimeout={600}
            transitionEnter={true}
            transitionLeave={false}
            className="column__content"
          >
            {focused && column.focus.align === 'left' && (
              <FocusedCardComponent
                isAdmin={isAdmin}
                key={focused.id}
                boardUrl={boardUrl}
                focused={focused}
                className="component--large"
                showVotes={column.voting.displayed}
              />
            )}

            {!isExtended && (
              <StackComponent
                boardUrl={boardUrl}
                cards={cards}
                isVotingAllowed={column.voting.enabled}
                isVoteSummaryShown={column.voting.displayed}
                column={column}
                isActive={isActive || !isCompactView}
                className={classNames('column__stack-component', {
                  ['column__stack-component--hidden']: Boolean(focused)
                })}
              />
            )}

            {focused && column.focus.align === 'right' && (
              <FocusedCardComponent
                isAdmin={isAdmin}
                key={focused.id}
                boardUrl={boardUrl}
                focused={focused}
                className="component--large"
                showVotes={column.voting.displayed}
              />
            )}
          </ReactCSSTransitionGroup>
        </div>
      </>
    );
  }
}

export default connect<StateColumnProps, null, OwnColumnProps>(mapStateToProps)(
  Column
);
