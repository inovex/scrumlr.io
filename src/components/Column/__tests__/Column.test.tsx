import {render} from "@testing-library/react";
import Column from "../Column";

const createColumn = () => <Column id="TestID" name="Testheader" color="planning-pink" />;

describe("Column", () => {
  describe("should render correctly", () => {
    test("column has correct accent-color", () => {
      const {container} = render(createColumn());
      expect(container.firstChild).toHaveClass("column accent-color__planning-pink");
    });

    test("column content is present", () => {
      const {container} = render(createColumn());
      expect(container.querySelector(".column")!.firstChild).toHaveClass("column__content");
    });

    test("column header is present", () => {
      const {container} = render(createColumn());
      expect(container.querySelector(".column__content")!.firstChild).toHaveClass("column__header");
    });

    test("column header title is present", () => {
      const {container} = render(createColumn());
      expect(container.querySelector(".column__header")!.firstChild).toHaveClass("column__header-title");
    });

    test("column header text is present", () => {
      const {container} = render(createColumn());
      expect(container.querySelector(".column__header-title")!.firstChild).toHaveClass("column__header-text");
    });

    test("column header cardnumber is present", () => {
      const {container} = render(createColumn());
      expect(container.querySelector(".column__header-title")!.lastChild).toHaveClass("column__header-card-number");
    });

    test("header text has correct title", () => {
      const {container} = render(createColumn());
      expect(container.firstChild).toHaveTextContent("Testheader");
    });
  });

  describe("should have correct style", () => {
    test("show column with correct style", () => {
      const {container} = render(createColumn());
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
