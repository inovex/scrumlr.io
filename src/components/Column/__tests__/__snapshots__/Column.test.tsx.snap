// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column should have correct style show column with correct style 1`] = `
<section
  class="column accent-color__planning-pink"
>
  <div
    class="column__content"
  >
    <header
      class="column__header"
    >
      <div
        class="column__header-title"
      >
        <h2
          class="column__header-text"
        >
          Testheader
        </h2>
        <span
          class="column__header-card-number"
        >
          0
        </span>
      </div>
      <div
        class="MuiInputBase-root MuiFilledInput-root NoteInput-root-15 MuiFilledInput-underline MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
      >
        <input
          class="MuiInputBase-input MuiFilledInput-input NoteInput-input-16 MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
          placeholder="Add your note..."
          type="text"
          value=""
        />
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
        >
          <svg
            class="note-input__icon"
          >
            icon-add.svg
          </svg>
        </div>
      </div>
    </header>
    <div
      class="column__notes-wrapper"
    >
      <ul
        class="column__note-list"
      />
    </div>
  </div>
</section>
`;
